src/Nokia_LCD.cpp

```diff
@@ -229,14 +229,24 @@ bool Nokia_LCD::printCharacter(char character) {
 bool Nokia_LCD::draw(const unsigned char bitmap[],
                      const unsigned int bitmap_size,
                      const bool read_from_progmem) {
+    
+    return Nokia_LCD::draw(bitmap, bitmap_size, kTotal_columns, read_from_progmem);
+}
+
+bool Nokia_LCD::draw(const unsigned char bitmap[],
+                     const unsigned int bitmap_size,
+                     const unsigned int bitmap_width,
+                     const bool read_from_progmem) {
     bool out_of_bounds = false;
+    const unsigned int currentX = mX_cursor;
     for (unsigned int i = 0; i < bitmap_size; i++) {
         unsigned char pixel =
             read_from_progmem ? pgm_read_byte_near(bitmap + i) : bitmap[i];
         if (mInverted) {
             pixel = ~pixel;
         }
-        out_of_bounds = sendData(pixel) || out_of_bounds;
+        send(pixel, true, false);
+        out_of_bounds = updateCursorPosition(currentX, bitmap_width) || out_of_bounds; 
```

Comment:
No need to have an extra variable, `mX_cursor` can be used.

Reply:
I need, because `mX_cursor` will change its value while we are drawing an image. When `mX_cursor` reaches the image width, we need to return it to the original column, it's `currentX`.

Reply:
Imagine you have an image of 50px width and 24px height (3 lines) and you want to align it at right. So, you start drawing at column 34, when you finish the first line at column 84, mX_cursor needs to be 34 again otherwise the image will be distorted. That's why I need currentX (I'm renaming it to x_start_position), I save it before starting the for loop.

Reply:
Right, now I notice it's outside the for loop. Perhaps call it `initialX` or something?
Also I finally got why you are adding the two positions together. :) 👍